go/libkb/context.go

```diff
@@ -390,8 +390,8 @@ func (m MetaContext) SwitchUserToActiveDevice(n NormalizedUsername, ad *ActiveDe
 	if n.IsNil() {
 		return nil
 	}
-	if err = n.CheckValid(); err != nil {
-		return err
+	if !n.IsValid() {
+		return NewBadUsernameError(n.String())
```

Comment:
Can the BadUsernameError be located in `kbun`. Seems weird to have to wrap it each call.

Reply:
I originally had that, but decided against it for a couple of reasons:

1) `BadUsernameError` isn't used just for bad keybase usernames, but also for bad twitter usernames, and also mismatches between an expected username and actual username. So it seems weird for a "generic" username error to live in `kbun`, and I didn't want to bother with splitting up the errors.
2) There's only one place (here) where we have to do the wrapping.

Reply:
Great that makes sense